### ISSUE TITLE
[Bug]: Multiplayer lobby - server details - NewGRF Settings

### ISSUE THREAD

**Issue by TheDude-gh** [OpenTTD/OpenTTD#9512]
### Version of OpenTTD

12.0-beta2

### Expected result

NewGRF Settings windows lists all NewGRFs on the server with proper GRF names.

### Actual result

Some NewGRFs names are displayed as "unknown". "Find missing content" function finds correct names though.

Probably happens for pre 12.x servers only.

### Steps to reproduce

Find a pre 12.x server in lobby with NewGRFs and click on NewGRF Settings. There is a high chance there will be "unknown" GRF.

**Comment by TrueBrain** [comment]
Indeed, this only happens for pre 12.x servers, and it is not something we will fix. Let me explain :)

Technical, what is happening: for the new multiplayer stuff, we had to drop the UDP protocol. In the old days UDP was used to exchange this information. Result: new clients cannot fetch this information from old servers.
It is not a total loss, you can press the `Find missing content online`, and it most likely will tell you the NewGRF names (as you mentioned).

Additionally what we do, is use 12.x servers to fill in the blanks for older servers. So if any 12.x server uses a NewGRF, it also fills in the name for older servers that use that same NewGRF. So after 12.0 is released, we will slowly see less and less `unknown`s in older servers .. and less older servers in general, ofc.

Lastly, as you cannot join older servers anyway, there is not much value (but also not none) in knowing NewGRFs the server is running exactly. In most use-cases the `Find missing content online` button should be sufficient. But there are some use-cases left, ofc.

As this problem is transient (in general, after a release it takes a few weeks for old servers to mostly disappear) and given it is only for pre 12.x servers, we think this is not a real problem to have. Of course during this beta it feels a bit odd :)

In conclusion, going to close this issue for now; if you disagree or have other arguments, please do let us know! And tnx for the report :)

**Comment by TheDude-gh** [comment]
Thanks for explanation.

I actually indeed disagree with this aproach, but I kind of understand your reasoning. But still, I find it quite imperfect to show unknown instead of getting the real name.

Interestingly, when I played with the GC listing, it returned like ~4000 GRFs in the lookup table two days ago and I didn't see any unknown names there, but now it returns something about ~1000. Which is still a lot, since servers use some ~700 NewGRFs in average anyway, so it would seem to me, that you can get all used GRFs in the GC listing no problem.

It is pitty the whole UDP layer was removed. It is not much use in new 12.x, but still, it could have been left there for some legacy, and removed perhaps later in 13.x or whatever version will be next.

Another thing with UDP packets removed, the nice feature of company details is away. Do you maybe plan to add this in future with the current GC thingy? It was quite useful to check company details before joining server.

Another reason I do not like UDP packets removed is it was also useful to get server details with 3rd party applications. For example I have web based app to scan servers and make some statistics (https://www.novapolis.net/newgrfsAservers.php).

With new GC I will have to rewrite it a bit, which is ok, but with GC, I can't get proper NewGRF info from older versions. And with UDP queries I can't get info from newer versions, so it would have to mix both, since some servers are still running several years old versions.

I guess this issue can stay closed, but one thing that could at least be changed is, the former master server returns also IP addresses for 12.x servers, which is quite useless for pre 12.x versions. The pre 12.x lobby will display those as empty rows in lobby. 
Wouldn't it be better, if master server filtered new versions out and didn't send them at all?

**Comment by TrueBrain** [comment]
> 
> Interestingly, when I played with the GC listing, it returned like ~4000 GRFs in the lookup table two days ago and I didn't see any unknown names there, but now it returns something about ~1000. Which is still a lot, since servers use some ~700 NewGRFs in average anyway, so it would seem to me, that you can get all used GRFs in the GC listing no problem.

The more servers run 12.x, the more information we get. But when a server leaves, after a while we also expire the GRF info. This is why you see this information coming and going. This is also why I expect that when we leave beta, it becomes much less of an issue very quickly. But do let me know if that is not the case!

> 
> It is pitty the whole UDP layer was removed. It is not much use in new 12.x, but still, it could have been left there for some legacy, and removed perhaps later in 13.x or whatever version will be next.

Issue with "keeping legacy" is that we also have to support it. And if our clients are not using it, it is likely to break without us knowing. So I don't disagree, but it puts a huge burden on our side. One we currently cannot carry, I am afraid :(

> 
> Another thing with UDP packets removed, the nice feature of company details is away. Do you maybe plan to add this in future with the current GC thingy? It was quite useful to check company details before joining server.

No, we have no plans to bring this back. The main issue with this is that it was given before you entered the server password. This was often considered a bug or at least a miss-feature. With the dropping of the lobby window, we also are no longer using it. For server statistics it is better to use the admin port (which does still have this information) .. but yeah, you can no longer snoop of all public servers what the companies are named etc :D If you have a good use-case why we should bring this back, please do let us know; this is always a conversation :)

> 
> Another reason I do not like UDP packets removed is it was also useful to get server details with 3rd party applications. For example I have web based app to scan servers and make some statistics (https://www.novapolis.net/newgrfsAservers.php).

Ha, I knew someone was plotting these graphs, I just couldn't find the URL when I was looking for it the other day :P Nice :D

I would argue it becomes a bit easier to keep this information, as you now just have to ask the GC for a listing every N seconds (where N > 30, as servers only update the GC every 30 seconds). It contains all the useful information (I think). If there is anything missing, like it was with gamescripts, do let me know. Happy to add bits and pieces if that is useful :)

> 
> With new GC I will have to rewrite it a bit, which is ok, but with GC, I can't get proper NewGRF info from older versions. And with UDP queries I can't get info from newer versions, so it would have to mix both, since some servers are still running several years old versions.

Yup. Basically we put the burden from us to you, and I do realise that. I hope this is not too much work for you, and if you need any help, don't be afraid to reach out!

> 
> I guess this issue can stay closed, but one thing that could at least be changed is, the former master server returns also IP addresses for 12.x servers, which is quite useless for pre 12.x versions. The pre 12.x lobby will display those as empty rows in lobby.
> Wouldn't it be better, if master server filtered new versions out and didn't send them at all?

You are not wrong. That would be a bug on game-coordinator (https://github.com/OpenTTD/game-coordinator/issues/51). Will see if I can fix that soon :) Tnx!
